js/core/net/ip4-address.js

```diff
@@ -14,74 +14,63 @@
-};
-
-IP4Address.prototype.equals = function(that) {
-  return this.a === that.a && this.b === that.b &&
-         this.c === that.c && this.d === that.d;
-};
-
-IP4Address.prototype.and = function(that) {
-  return new IP4Address(this.a & that.a, this.b & that.b,
-                        this.c & that.c, this.d & that.d);
-};
-
-IP4Address.prototype.hash = function() {
-  return (this.a | (this.b << 8) | (this.c << 16) | (this.d << 24)) >>> 0;
-};
-
-IP4Address.prototype.isBroadcast = function() {
-  return 255 === this.a && 255 === this.b &&
-         255 === this.c && 255 === this.d;
-};
-
-IP4Address.prototype.isAny = function() {
-  return 0 === this.a && 0 === this.b &&
-         0 === this.c && 0 === this.d;
-};
-
-IP4Address.ANY = new IP4Address(0, 0, 0, 0);
-IP4Address.BROADCAST = new IP4Address(255, 255, 255, 255);
-IP4Address.LOOPBACK = new IP4Address(127, 0, 0, 1);
-
-IP4Address.parse = function(str) {
-  if (str instanceof IP4Address) {
-    return str;
+class IP4Address {
+  constructor(a, b, c, d) {
+    this.a = (a & 0xff) >>> 0;
+    this.b = (b & 0xff) >>> 0;
+    this.c = (c & 0xff) >>> 0;
+    this.d = (d & 0xff) >>> 0;
   }
-
-  if ('string' !== typeof str) {
-    return null;
+  toString() {
+    return `${this.a}.${this.b}.${this.c}.${this.d}`;
   }
-
-  var p = str.trim().split('.');
-  if (4 !== p.length) {
-    return null;
+  toInteger() {
+    return ((this.a << 24) | (this.b << 16) | (this.c << 8) | this.d) >>> 0;
   }
-
-  var a = new Array(4);
-  for (var i = 0; i < 4; ++i) {
-    var v = p[i] | 0;
-    if (v !== +p[i] || v < 0 || v > 255) {
-      return null;
+  equals(that) {
+    return this.a === that.a && this.b === that.b &&
+           this.c === that.c && this.d === that.d;
+  }
+  and(that) {
+    return new IP4Address(this.a & that.a, this.b & that.b,
+                          this.c & that.c, this.d & that.d);
+  }
+  hash() {
+    return (this.a | (this.b << 8) | (this.c << 16) | (this.d << 24)) >>> 0;
+  }
+  isBroadcast() {
+    return this.a === 255 && this.b === 255 &&
+           this.c === 255 && this.d === 255;
+  }
+  isAny() {
+    return this.a === 0 && this.b === 0 &&
+           this.c === 0 && this.d === 0;
+  }
+  static parse(str) {
+    if (str instanceof IP4Address) return str;
+    if (typeof str !== 'string') return null;
+
+    const p = str.trim().split('.');
+    if (p.length !== 4) return null;
+
+    const a = new Array(4);
+    for (let i = 0; i < 4; ++i) {
+      const v = p[i] | 0;
+      if (v !== +p[i] || v < 0 || v > 255) return null;
+      a[i] = v;
     }
 
-    a[i] = v;
+    return new IP4Address(a[0], a[1], a[2], a[3]);
   }
-
-  return new IP4Address(a[0], a[1], a[2], a[3]);
-};
+  static get ANY() {
+    return new IP4Address(0, 0, 0, 0);
+  }
+  static get BROADCAST() {
+    return new IP4Address(255, 255, 255, 255);
+  }
+  static get LOOPBACK() {
+    return new IP4Address(127, 0, 0, 1);
+  }
```

Comment:
would be cool to have those ip objects allocated once and cached, getters can be slower too I guess